central/alert/service/service_impl.go

```diff
@@ -62,14 +63,14 @@ var (
 		},
 	})
 
-	// groupByFunctions provides a map of functions that group slices of ListAlet objects by category or by cluser.
-	groupByFunctions = map[v1.GetAlertsCountsRequest_RequestGroup]func(*storage.ListAlert) []string{
-		v1.GetAlertsCountsRequest_UNSET: func(*storage.ListAlert) []string { return []string{""} },
-		v1.GetAlertsCountsRequest_CATEGORY: func(a *storage.ListAlert) (output []string) {
-			output = append(output, a.GetPolicy().GetCategories()...)
+	// groupByFunctions provides a map of functions that group slices of result objects by category or by cluster.
+	groupByFunctions = map[v1.GetAlertsCountsRequest_RequestGroup]func(result search.Result) []string{
+		v1.GetAlertsCountsRequest_UNSET: func(result search.Result) []string { return []string{""} },
+		v1.GetAlertsCountsRequest_CATEGORY: func(a search.Result) (output []string) {
+			output = append(output, a.Matches[search.Category.String()]...)
```

Comment:
The key in `Matches` is the field path in the object like `.policy.name` and sth.

You can get the field paths from the list alerts options map as such,
```
field, ok := {alertMappings}.OptionsMap.Get(search.Category.String())
fieldPath := field.GetFieldPath()
```

Reply:
Got it